src/drivers/intel/ssp/ssp.c

```diff
@@ -712,6 +712,52 @@ static int ssp_set_config(struct dai *dai, struct ipc_config_dai *common_config,
 	ssp->state[DAI_DIR_PLAYBACK] = COMP_STATE_PREPARE;
 	ssp->state[DAI_DIR_CAPTURE] = COMP_STATE_PREPARE;
 
+	if (config->flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS) {
```

Comment:
@plbossart With below change

```
/* DAI_CONFIG flags */
#define SOF_DAI_CONFIG_FLAGS_MASK	0x3
#define SOF_DAI_CONFIG_FLAGS_NONE	(0 << 0) /**< DAI_CONFIG sent without stage information */
#define SOF_DAI_CONFIG_FLAGS_HW_PARAMS	(1 << 0) /**< DAI_CONFIG sent during hw_params stage */
#define SOF_DAI_CONFIG_FLAGS_HW_FREE	(2 << 0) /**< DAI_CONFIG sent during hw_free stage */
#define SOF_DAI_CONFIG_FLAGS_RFU	(3 << 0) /**< not used, reserved for future use */
```
We are not supposed to use `config->flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS` I would suggest to use 
```
switch (config->flags & SOF_DAI_CONFIG_FLAGS_MASK)
        case SOF_DAI_CONFIG_FLAGS_HW_PARAMS:
...
```

Reply:
yeah, ```SOF_DAI_CONFIG_FLAGS``` looks more like an enum.

Reply:
I followed @lyakh's suggestion to use a 2-bit flag value. It's not wrong to use the mask, the results are correct since we don't use the value 3, but I will indeed change the code in case we ever use it.

Reply:
fixed